yjit/src/codegen.rs

```diff
@@ -7061,15 +7062,38 @@ fn iseq_get_return_value(iseq: IseqPtr, captured_opnd: Option<Opnd>, ci_flags: u
     }
     match first_insn {
         YARVINSN_getlocal_WC_0  => {
-            // Only accept simple positional only cases for both the caller and the callee.
+            // Accept only cases where only positional arguments are used by both the callee and the caller.
+            // Keyword arguments may be specified by the callee or the caller but not used.
             // Reject block ISEQs to avoid autosplat and other block parameter complications.
-            if captured_opnd.is_none() && unsafe { rb_simple_iseq_p(iseq) } && ci_flags & VM_CALL_ARGS_SIMPLE != 0 {
-                let ep_offset = unsafe { *rb_iseq_pc_at_idx(iseq, 1) }.as_u32();
-                let local_idx = ep_offset_to_local_idx(iseq, ep_offset);
-                Some(IseqReturn::LocalVariable(local_idx))
-            } else {
-                None
+            if captured_opnd.is_some()
+                // Reject if block ISEQ is present
+                || block.is_some()
+                // Equivalent to `VM_CALL_ARGS_SIMPLE - VM_CALL_KWARG - has_block_iseq`
+                || ci_flags & (
+                      VM_CALL_ARGS_SPLAT
+                    | VM_CALL_KW_SPLAT
+                    | VM_CALL_ARGS_BLOCKARG
+                    | VM_CALL_FORWARDING
```

Comment:
Since overall we're targeting unused keyword parameters, allowing the call site to specify keywords seems odd. I don't see checks for passing an unknown keyword, and I wonder if they are rejected as they should be.

Reply:
Good catch! I tested this locally but I forgot to add a test case for it. Added one in 41276418b4073e4d01cbe2da051e2c2758495668.

Reply:
The reason I chose to allow keyword arguments at the call site is to allow calls like the following to be inlined:

```ruby
def let(value, type, checked: true)
    value
end

# ...

let(5, Integer, checked: false)
```

Reply:
I have to leave now but I will add this later this afternoon.